From 90a869635fe5481b97c112a2635aece04a91bcc1 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 20 Dec 2018 15:28:26 +0100
Subject: [PATCH 0027/1010] kasan: irq_pipeline: serialize reports from oob
 contexts

KASAN may detect issues in code running out-of-band, so we have to
serialize the report handler accordingly by turning the regular
spinlock into a hard lock.

This will trigger massive a latency spot when a report is emitted, but
since kernel sanity is at stake when this happens, such spike is not
the most urgent issue.

This also means that the KASAN report lock becomes a raw lock -
therefore cannot sleep with full preemption enabled.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 mm/kasan/report.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/kasan/report.c b/mm/kasan/report.c
index 240560ec8d..2f01a18f6e 100644
--- a/mm/kasan/report.c
+++ b/mm/kasan/report.c
@@ -173,7 +173,7 @@ static void update_kunit_status(bool sync)
 static void update_kunit_status(bool sync) { }
 #endif
 
-static DEFINE_RAW_SPINLOCK(report_lock);
+static DEFINE_HARD_SPINLOCK(report_lock);
 
 static void start_report(unsigned long *flags, bool sync)
 {
-- 
2.30.2

